sdk/core/src/gateway/gateway_proxy.rs

```diff
@@ -1,17 +1,23 @@
 use reqwest::Client;
 
+pub const CHAIN_SIMULATOR_GATEWAY: &str = "http://localhost:8085";
+
 /// Allows communication with the MultiversX gateway API.
 #[derive(Clone, Debug)]
 pub struct GatewayProxy {
     pub(crate) proxy_url: String,
     pub(crate) client: Client,
+    pub chain_simulator: bool,
 }
 
 impl GatewayProxy {
     pub fn new(proxy_url: String) -> Self {
+        let chain_simulator = proxy_url == CHAIN_SIMULATOR_GATEWAY;
```

Comment:
We should allow callers to set their own URL, who knows, maybe at some point it will be running on a different port.

We should pass the URL from the interactor